plugin/type_inet/sql_type_inet.cc

```diff
@@ -499,8 +499,10 @@ size_t Inet6::to_string(char *dst, size_t dstsize) const
       // hex-letters without leading zeros (recommended IPv6-format).
       //
       // If it is not the last field, append closing ':'.
-
-      p += sprintf(p, "%x", ipv6_words[i]);
+      dstsize_available= dstend - p;
+      int res= snprintf(p, dstsize_available, "%x", ipv6_words[i]);
+      DBUG_ASSERT(res >= 0);
+      p += res;
```

Comment:
Should `dstsize_available` be decreased here as well? E.g. `p += res; dstsize_available -= res;`

Reply:
Actually it doesn't have to be set at all since it is only read once per block, I removed line 502 here. The other blocks do not update `dstsize_available` after doing the pointer math.

Reply:
Must have just been flaky, all required tests pass now